src/common/pf_alarm.c

```diff
@@ -2731,19 +2731,45 @@ int pf_alarm_send_diagnosis (
          }
       }
 
-      // TODO Calculate module_ident, submodule_ident
-      // The ModuleIdentNumber of the portsubmodule which is connected to device
-      // ‘b’ The SubModuleIdentNumber of the portsubmodule which is connected to
-      // device ‘b’
+      ret = pf_cmdev_get_module_ident (net, api_id, slot_nbr, &module_ident);
+      if (ret != 0)
+      {
+         LOG_ERROR (
+            PF_ALARM_LOG,
+            "Alarm(%d): Failed to get module ident for slot: %u",
+            __LINE__,
+            slot_nbr);
+      }
```

Comment:
Should we `return ret; `at error? Also below.

Reply:
Updated so return code is -1 also if operation fails to get the module ids.
Done